### PR TITLE
feat: Add maxLength and autoSize in Typography editable config

### LINKED ISSUE #23801: Add maxLength to Editable Typography
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
I am using the editable typography in my code in a notes app and I need to have a maximum number of characters allowed set. This is not possible right now. The way I do it right now is I check the length of the new string when onChange is fired but that is too late as I cannot give users live feedback. Someone might right 1000 characters and when try to save it, it will say max characters is 500. 

### What does the proposed API look like?
Editable uses Textarea therefore we can easily pass a prop inside the editable object to the Base component and then Base component can pass it down to Editable.

### PR DESCRIPTION
**Pull request by CornerSkyless**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

希望在 Typography 的 editable 中传入 maxLength 属性来控制 Textarea 
[Issue链接](https://github.com/ant-design/ant-design/issues/23801)

close https://github.com/ant-design/ant-design/issues/23801

### 💡 Background and solution

增加了 maxLength 属性并且传入了调用的 Textarea 中，顺便把 autoSize 属性也加上了

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  feat: Add  maxLength & autoSize in Typography editable config |
| 🇨🇳 Chinese |  Typography 的 editable 配置中增加了 maxLength & autoSize 支持   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/typography/index.en-US.md](https://github.com/CornerSkyless/ant-design/blob/add_typography_max_length/components/typography/index.en-US.md)
[View rendered components/typography/index.zh-CN.md](https://github.com/CornerSkyless/ant-design/blob/add_typography_max_length/components/typography/index.zh-CN.md)

### CODE REVIEW

File: components/typography/index.zh-CN.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/GOM1KQ24O/Typography.svg
 | copyable | 是否可拷贝，为对象时可进行各种自定义 | boolean \| { text: string, onCopy: function, icon: ReactNode, tooltips: \[ReactNode, ReactNode\] } | false | `icon` 和 `tooltips` 在 `4.4.0` 支持 |
 | delete | 添加删除线样式 | boolean | false |  |
 | disabled | 禁用文本 | boolean | false |  |
-| editable | 是否可编辑，为对象时可对编辑进行控制 | boolean \| { editing: boolean, onStart: function, onChange: function(string) } | false |  |
+| editable | 是否可编辑，为对象时可对编辑进行控制 | boolean \| { editing: boolean, maxLength: number, autoSize: true \| false \| { minRows: number, maxRows: number }, onStart: function, onChange: function(string) } | false |  |
```
Comment (xrkffgg) [inline components/typography/index.zh-CN.md:27]:
```suggestion
| editable | 是否可编辑，为对象时可对编辑进行控制 | boolean \| { editing: boolean, maxLength: number, autoSize: true \| false \| { minRows: number, maxRows: number }, onStart: function, onChange: function(string) } | false | maxLength: 4.5.0 \| autoSize: 4.5.0  |
```

其他的也加一下

**Comment by xrkffgg** [comment]
可以在 可交互 的 Demo 中加一个例子
- https://ant.design/components/typography-cn/#components-typography-demo-interactive

File: components/typography/index.zh-CN.md (hunk 1 above)
Comment (CornerSkyless) [reply components/typography/index.zh-CN.md]: 加上了🙆‍♂️

File: components/typography/index.zh-CN.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/GOM1KQ24O/Typography.svg
 | copyable | 是否可拷贝，为对象时可进行各种自定义 | boolean \| { text: string, onCopy: function, icon: ReactNode, tooltips: \[ReactNode, ReactNode\] } | false | `icon` 和 `tooltips` 在 `4.4.0` 支持 |
 | delete | 添加删除线样式 | boolean | false |  |
 | disabled | 禁用文本 | boolean | false |  |
-| editable | 是否可编辑，为对象时可对编辑进行控制 | boolean \| { editing: boolean, onStart: function, onChange: function(string) } | false |  |
+| editable | 是否可编辑，为对象时可对编辑进行控制 | boolean \| { editing: boolean, maxLength: number, autoSize: true \| false \| { minRows: number, maxRows: number }, onStart: function, onChange: function(string) } | false | `maxLength` 和 `autoSize` 在 `4.4.5` 支持 |
```
Comment (xrkffgg) [inline components/typography/index.zh-CN.md:27]: 版本这里简单一点就行，和英文一样吧

Comment (CornerSkyless) [reply components/typography/index.zh-CN.md]: 之前我看到上面 icon 和 tooltips 那边写的比较全就复制那边的了 🤣 现在改好了~

File: components/typography/index.zh-CN.md (hunk 1 above)
Comment (xrkffgg) [reply components/typography/index.zh-CN.md]: 4.5.0  Not 4.4.5
Feature 会发 Minor 版本

Comment (xrkffgg) [reply components/typography/index.zh-CN.md]: @CornerSkyless

**Comment by xrkffgg** [comment]
4.5.0 Not 4.4.5
Feature 会发 Minor 版本

**Comment by CornerSkyless** [comment]
改成 4.5.0 了👌

File: components/typography/index.zh-CN.md (hunk 1 above)
Comment (CornerSkyless) [reply components/typography/index.zh-CN.md]: 额我刚刚复制错了 = =

**Comment by 07akioni** [comment]
测试挂了

**Comment by CornerSkyless** [comment]
> 测试挂了

之前忘记加 snapshot 了，现在加上了

**Comment by CornerSkyless** [comment]
@07akioni 那个，我想问一下，我 lint 的时候发现报错的代码不是我这边的，测试不能通过是要怎么处理呀...🤕

**Comment by yoyo837** [comment]
先rebase一遍看看

**Comment by xrkffgg** [comment]
4.5.0 版本已发。你这次没赶上。
改成 4.6.0 吧

**Review by zombieJ — APPROVED** [review]

**Pull request merged**